core/zoe/zoe.chainmail

```diff
@@ -0,0 +1,163 @@
+   */
+  getSeatIssuer() -> (Issuer);
+
+  /**
+   * Get the long-lived escrowReceiptIssuer. The mint associated
+   * with the escrowReceiptIssuer creates the ERTP payments that
+   * represent proof of escrow and may be passed into a
+   * governingContract method.
+   */
+  getEscrowReceiptIssuer() -> (Issuer);
+
+}
+
+interface ZoeUserFacet ( ) {
+
+  /** 
+   * To escrow, the user must provide an `offerDesc` which is a list of
+   * objects with a rule property (such as 'haveExactly') and an amount
+   * property. The objects in the `offerDesc` must be in the same order
+   * as the issuers array. That is, the amount in index 0 of `offerDesc`
+   * should be an amount for the issuer in index 0 of the issuers array.
+   * `payments` is an array of the actual payments to be escrowed,
+   * following the rules in the offerDesc. If the offerDesc rule is
+   * 'haveExactly' or 'haveAtMost', then a payment matching the amount is
+   * expected. If the offerDesc rule is 'wantAtLeast' or 'wantExactly'
+   * then the payment will be ignored and should be undefined.
+   */
+  escrow (offerDesc :List, payments :List) -> (escrowReceiptAndClaimPayoff);
+
+  /**
+   * Returns the array of issuers that the zoeInstance was instantiated
+   * with. This is helpful in the case that the user has forgotten the order.
+   */
+  getIssuers() -> (List);
+
+}
+
+interface ZoeGoverningContractFacet () {
+
+  /** 
+   * Instruct Zoe to try reallocating for the given offerIds.
+   * Reallocation is a matrix (array of arrays) where the rows are the
+   * quantities to be paid to the player who made the offer at the same
+   * index in the offerIds array. The reallocation will only happen if
+   * offer safety and conservation of rights are true, as enforced by Zoe.
+   */
+  reallocate (offerIds :List, reallocation :List);
+
+  /** 
+   * Eject the offer, taking the current allocation for that offer and
+   * creating payments to be returned to the user. The payments can be
+   * claimed by the user by taking the `claimPayment` ERTP payment
+   * received after escrowing, calling `unwrap` and calling the `claimPayoff`
+   * method on the resulting use object. No offer safety checks are
+   * done here because any previous reallocation performed those checks.
+   */
+  eject (offerIds :List);
+
+  /**
+   * Create an empty offer for recordkeeping purposes (Autoswap uses
+   * this to create the liquidity pool). 
+   */
+  escrowEmptyOffer ( ) -> (offerId);
+  
+  /**
+   * Escrow an offer created by the governing contract. Autoswap uses
+   * this to mint liquidity tokens and add them to the rights managed by
+   * the zoeInstance.
+   */
+  escrowOffer (offerDesc :List, offerPayments :List) -> (offerId);
+
+  /** 
+   * A helper function provided by Zoe to allow the governing contract
+   * to easily burn and validate an escrowReceipt received from the
+   * user. 
+   */
+  burnEscrowReceipt (escrowReceipt :Payment) -> (Quantity);
```

Comment:
I see several places  (here and [elsewhere](https://github.com/Agoric/ERTP/pull/136/files)) where the result of `burnEscrowReceipt()` is destructured as `{ id, offerMade }`, and then `id` is used as an index into offers. Why does this work?

Reply:
ah! Because the quantity is an object with the properties { id, offerMade }. I had to double check that just now because it didn't make sense to me either.